test/Microsoft.AspNetCore.Razor.Language.Test/Legacy/TagHelperParseTreeRewriterTest.cs

```diff
@@ -1108,6 +1108,91 @@ public static TheoryData AllowedChildrenData
             EvaluateData(descriptors, documentContent, (MarkupBlock)expectedOutput, (RazorDiagnostic[])expectedErrors);
         }
 
+        [Fact]
+        public void Rewrite_AllowsHtmlCommentsAsChildren()
```

Comment:
vars in this file, can apply general comments in this method to the one below as well.